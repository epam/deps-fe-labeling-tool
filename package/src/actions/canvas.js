import { createAction } from 'redux-actions'

const FEATURE_NAME = 'CANVAS'

const setZoom = createAction(
  `${FEATURE_NAME}/SET_ZOOM`
)

const setRotationAngle = createAction(
  `${FEATURE_NAME}/SET_ROTATION_ANGLE`,
  (page, angle) => ({
    page,
    angle
  })
)

const setScale = createAction(
  `${FEATURE_NAME}/SET_SCALE`
)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

export {
  resetDefault,
  setZoom,
  setScale,
  setRotationAngle
}
