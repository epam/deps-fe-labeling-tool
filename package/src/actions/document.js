import { createAction } from 'redux-actions'

const FEATURE_NAME = 'DOCUMENT'

const storeDocument = createAction(`${FEATURE_NAME}/STORE`)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

export {
  resetDefault,
  storeDocument
}
