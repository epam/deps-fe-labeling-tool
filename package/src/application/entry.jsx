import 'antd/lib/style/themes/default.less'
import 'antd/lib/style/core/index.less'
import React from 'react'
import ReactDOM from 'react-dom'
import { LabelingTool } from '@/application/LabelingTool'
import { SHIM_CONFIG } from '@/config/shim'

const LABELING_TOOL_EXPORT_KEY = 'LabelingTool'

const mount = (targetId, config) => {
  ReactDOM.render(
    <LabelingTool config={config} />,
    document.getElementById(targetId)
  )
}

const unmount = (targetId) => ReactDOM.unmountComponentAtNode(
  document.getElementById(targetId)
)

if (window.LABELING_TOOL_SHIM_TARGET_ID) {
  mount(window.LABELING_TOOL_SHIM_TARGET_ID, SHIM_CONFIG)
} else {
  window[LABELING_TOOL_EXPORT_KEY] = {
    mount,
    unmount
  }
}
