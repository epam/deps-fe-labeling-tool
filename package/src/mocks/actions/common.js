const resetDefault = jest.fn(() => ({ type: 'MOCK_RESET_DEFAULT' }))

const mockCommonActions = {
  resetDefault
}

export {
  mockCommonActions
}
