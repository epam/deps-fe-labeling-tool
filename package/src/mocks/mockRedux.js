const mockStore = {
  MOCK_STORE: 'MOCK_STORE',
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({
    MOCK_STATE: 'MOCK_STATE'
  }))
}

const mockRedux = {
  createStore: jest.fn(() => mockStore),
  applyMiddleware: jest.fn(() => mockStore),
  combineReducers: jest.fn(() => ({
    MOCK_COMBINED_REDUCER: 'MOCK_COMBINED_REDUCER'
  }))
}

export {
  mockRedux
}
