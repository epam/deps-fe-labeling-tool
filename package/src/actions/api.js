import { createAction } from 'redux-actions'
import { rotationAnglesSelector } from '@/selectors/canvas'
import { pageImageUrlSelector } from '@/selectors/document'
import { markupSelector } from '@/selectors/markup'
import { fieldsSelector } from '@/selectors/model'
import { primaryLanguageSelector } from '@/selectors/ocr'
import { getApi } from '@/services/api'

const FEATURE_NAME = 'API'

const requestAttempt = createAction(
  `${FEATURE_NAME}/REQUEST_ATTEMPT`
)

const requestSuccess = createAction(
  `${FEATURE_NAME}/REQUEST_SUCCESS`
)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

const requestFailure = createAction(
  `${FEATURE_NAME}/REQUEST_FAILURE`,
  (requestId, error) => ({
    requestId,
    error
  })
)

export const createRequestAction = (requestId, actionCreator) => {
  const requestActionCreator = (...args) => async (dispatch, getState) => {
    try {
      dispatch(requestAttempt(requestId))
      const result = await actionCreator(...args)(dispatch, getState)
      dispatch(requestSuccess(requestId))
      return result
    } catch (error) {
      dispatch(requestFailure(requestId, error.message))
      throw error
    }
  }
  requestActionCreator.toString = () => requestId
  return requestActionCreator
}

const save = createRequestAction(
  'save',
  () => (dispatch, getState) => {
    const state = getState()
    const markup = markupSelector(state)
    const rotationAngles = rotationAnglesSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().save(markup, rotationAngles, language)
  }
)

const saveMarkup = createRequestAction(
  'saveMarkup',
  () => (dispatch, getState) => {
    const state = getState()
    const markup = markupSelector(state)
    const rotationAngles = rotationAnglesSelector(state)
    const language = primaryLanguageSelector(state)
    const fields = fieldsSelector(state)
    return getApi().saveMarkup(markup, rotationAngles, language, fields)
  }
)

const ocrText = createRequestAction(
  'ocrText',
  (engine, coordinates) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().ocrText(
      engine,
      pageImageUrl,
      coordinates,
      language
    )
  }
)

const omrArea = createRequestAction(
  'omrArea',
  (coordinates) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)

    return getApi().omrArea(
      pageImageUrl,
      coordinates
    )
  }
)

const detectTables = createRequestAction(
  'detectTables',
  (selectedAreaCoords) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    return getApi().detectTables(
      pageImageUrl,
      selectedAreaCoords
    )
  }
)

const ocrTable = createRequestAction(
  'ocrTable',
  (engine, table) => (dispatch, getState) => {
    const state = getState()
    const pageImageUrl = pageImageUrlSelector(state)
    const language = primaryLanguageSelector(state)
    return getApi().ocrTable(
      engine,
      pageImageUrl,
      table,
      language
    )
  }
)

export {
  detectTables,
  ocrTable,
  ocrText,
  omrArea,
  resetDefault,
  requestAttempt,
  requestSuccess,
  requestFailure,
  save,
  saveMarkup
}
