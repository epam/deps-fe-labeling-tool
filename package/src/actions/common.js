import { resetDefault as resetApi } from './api'
import { resetDefault as resetCanvas } from './canvas'
import { resetDefault as resetDocument } from './document'
import { resetDefault as resetImage } from './image'
import { resetDefault as resetMarkup } from './markup'
import { resetDefault as resetModel } from './model'
import { resetDefault as resetOcr } from './ocr'
import { resetDefault as resetPagination } from './pagination'
import { resetDefault as resetSettings } from './settings'
import { resetDefault as resetTools } from './tools'

const resetDefault = () => (dispatch) => {
  dispatch(resetApi())
  dispatch(resetOcr())
  dispatch(resetCanvas())
  dispatch(resetImage())
  dispatch(resetDocument())
  dispatch(resetMarkup())
  dispatch(resetPagination())
  dispatch(resetTools())
  dispatch(resetModel())
  dispatch(resetSettings())
}

export {
  resetDefault
}
