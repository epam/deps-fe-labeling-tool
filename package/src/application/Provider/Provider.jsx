import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { Provider as StoreProvider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunk from 'redux-thunk'
import { resetDefault } from '@/actions/common'
import { rootReducer } from '@/reducers/root'

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)))

const Provider = (props) => {
  useEffect(() => () => {
    store.dispatch(resetDefault())
  })

  return (
    <StoreProvider store={store} >
      {props.children}
    </StoreProvider>
  )
}

Provider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export {
  Provider
}
