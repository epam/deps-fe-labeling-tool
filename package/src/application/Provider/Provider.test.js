import { mockCommonActions } from '@/mocks/actions/common'
import { mockRedux } from '@/mocks/mockRedux'
import React from 'react'
import { render } from '@testing-library/react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { createStore } from 'redux'
import { resetDefault } from '@/actions/common'
import { Provider } from '@/application/Provider'

jest.mock('redux', () => mockRedux)

jest.mock('@/actions/common', () => mockCommonActions)

describe('Container: Provider', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(
      <Provider>
        <div />
      </Provider>
    )
  })

  it('should render correct layout', () => {
    const shallowRenderer = new ShallowRenderer()
    const wrapper = shallowRenderer.render(
      <Provider>
        <div />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should call store.dispatch with resetDefault on unmount', () => {
    const store = createStore()
    wrapper.unmount()
    expect(store.dispatch).nthCalledWith(1, resetDefault())
  })
})
