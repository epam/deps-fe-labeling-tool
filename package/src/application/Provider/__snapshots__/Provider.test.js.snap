// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: Provider should render correct layout 1`] = `
<Provider
  store={
    Object {
      "MOCK_STORE": "MOCK_STORE",
      "dispatch": [MockFunction],
      "getState": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "MOCK_STATE": "MOCK_STATE",
            },
          },
          Object {
            "type": "return",
            "value": Object {
              "MOCK_STATE": "MOCK_STATE",
            },
          },
        ],
      },
      "subscribe": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
>
  <div />
</Provider>
`;
